common/sketch/src/main/java/org/apache/spark/util/sketch/DefaultBloomFilter.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util.sketch;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+
+public class DefaultBloomFilter extends BloomFilter {
+
+  private final int numHashFunctions;
+  private final BitArray bits;
+
+  public DefaultBloomFilter(int numHashFunctions, long numBits) {
+    this.numHashFunctions = numHashFunctions;
+    this.bits = new BitArray(numBits);
+  }
+
+  @Override
+  public double expectedFpp() {
+    return Math.pow((double) bits.bitCount() / bits.bitSize(), numHashFunctions);
+  }
+
+  @Override
+  public long bitSize() {
+    return bits.bitSize();
+  }
+
+  private static long hashObjectToLong(Object item) {
```

Comment:
is this shared with count-min sketch? maybe we can put this into a Util class used by both.